portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -20,36 +20,80 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-import java.util.ArrayList;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+ private ArrayList<String> comment= new ArrayList<String>();
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    
-    ArrayList<String> comment= new ArrayList<String>();
-    comment.add("Hello!");
-    comment.add("Hey!");
-    comment.add("Bye!");
+    Query query = new Query("Comment") ;
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
New object is being created here for each request, rather you can make this field static and initialize it in init() method so that object will be created once and shared across requests.
Won't uplift performance very much here but in practice try having static objects which can be shared across threads.